lib/runtime-core/src/module.rs

```diff
@@ -173,6 +326,107 @@ impl Clone for Module {
     }
 }
 
+/// The type of import. This indicates whether the import is a function, global, memory, or table.
+// TODO: discuss and research Kind vs Type;
+// `Kind` has meaning to me from Haskell as an incomplete type like
+// `List` which is of kind `* -> *`.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ImportType {
+    /// The import is a function.
+    // TODO: why does function have no data?
+    Function,
+    /// The import is a global variable.
+    Global {
+        /// Whether or not the variable can be mutated.
+        mutable: bool,
+        /// The Wasm type that the global variable holds.
+        // TODO: attempt to understand explanation about 128bit globals:
+        // https://github.com/WebAssembly/simd/blob/master/proposals/simd/SIMD.md#webassembly-module-instatiation
+        type_: Type,
```

Comment:
I just realized of this. I think using `ty` might be a better field name :)

Reply:
We can also do `r#type` if we want a field named type, but I think it's better in general to not use that syntax because it might confuse people not familiar with it

Reply:
(`r#` makes it literal and allows you to use any name, including reserved keywords.  It was added to allow adding new keywords into the language and having a clear migration path for old code to not break.  It can also be used to not have to name things weirdly, but it requires the syntax to use)